Added sample code - ODSA for Java Devs - Spring Boot 3 + Data JPA P2

juarezjuniorgithub opened the pull request:
@Kuassim 
Signed-off-by: Juarez Barbosa Junior juarez.barbosa@gmail.com

Kuassim commented on java/odsa-spring/src/main/resources/application.properties:
```diff
@@ -0,0 +1,21 @@
+spring.datasource.driver-class-name=oracle.jdbc.OracleDriver
+spring.datasource.type=oracle.ucp.jdbc.PoolDataSource
+spring.datasource.oracleucp.connection-pool-name=connectionPoolName1
+spring.datasource.oracleucp.initial-pool-size=15
```
Do you need that many initial connections for this demo use case?

juarezjuniorgithub replied:
Hi Kuassi, spot on! Not at all, I copied that verbatim from your OCW/J1 deck. Changed it to the values below as this is for educational purposes only indeed, so it's fixed now. Thanks!

spring.datasource.oracleucp.initial-pool-size=1
spring.datasource.oracleucp.min-pool-size=1
spring.datasource.oracleucp.max-pool-size=2

Kuassim commented on java/odsa-spring/src/main/java/com/oracle/dev/jdbc/odsa/spring/Employee.java:
```diff
@@ -0,0 +1,115 @@
+import jakarta.persistence.Table;
+
+@Entity
+@Table(name = "EMPLOYEE", schema = "ADMIN")
```
The ADMIN schema should not be used for user/developer resources. As our code samples show the best practices, I'd suggest using a different schema. There use to be an HR/HR schema for demo purposes but I am not sure it's still around on Autonomous DB. If not you can ask the user to create one using SQL Developer Web interface (available on ADB console).

juarezjuniorgithub replied:
fixed

gvenzl commented on java/odsa-spring/script/ODSASpringBoot.sql:
```diff
@@ -0,0 +1,33 @@
+CREATE USER ODSA_USER IDENTIFIED BY <YOUR_PASSWORD>;
+GRANT DWROLE TO ODSA_USER;
+GRANT CREATE SESSION TO ODSA_USER;
```
This is redundant, as it is already provided by the `DWROLE`, see the [documentation](https://docs.oracle.com/en/cloud/paas/autonomous-database/adbsa/manage-users-privileges.html#GUID-50450FAD-9769-4CF7-B0D1-EC14B465B873).

juarezjuniorgithub replied:
This is fixed as well, thanks again for your support and the orientation/documentation.

gvenzl commented on java/odsa-spring/src/main/resources/application.properties:
```diff
@@ -0,0 +1,21 @@
+spring.datasource.oracleucp.connection-factory-class-name=oracle.jdbc.pool.OracleDataSource
+
+# Enable JPA / Hibernate debugging
+spring.jpa.show-sql=true
+spring.jpa.properties.hibernate.format_sql=true
+logging.level.org.hibernate.SQL=DEBUG
+logging.level.org.hibernate.type.descriptor.sql.BasicBinder=TRACE
```
Do we really need debugging for a demo script? One would assume that a developer understands when/how to turn on debugging.

juarezjuniorgithub replied:
adjustment accepted, debugging properties removed. @Kuassim please review, thanks

juarezjuniorgithub commented on java/odsa-spring/script/ODSASpringBoot.sql:
```diff
@@ -0,0 +1,33 @@
+CREATE USER ODSA_USER IDENTIFIED BY <YOUR_PASSWORD>;
+GRANT DWROLE TO ODSA_USER;
+GRANT CREATE SESSION TO ODSA_USER;
+GRANT UNLIMITED TABLESPACE TO ODSA_USER;
```
That's fixed, thanks for your support and orientation.

juarezjuniorgithub commented on java/odsa-spring/pom.xml:
```diff
@@ -0,0 +1,108 @@
+				<artifactId>azure-webapp-maven-plugin</artifactId>
+				<version>2.5.0</version>
+				<configuration>
+					<subscriptionId>cc5cb530-ce48-488b-a1a7-5a1fec12ed5e</subscriptionId>
```
Fixed

juarezjuniorgithub commented:
@Kuassim @jeandelavarene the changes were performed as required - fix indentation.
Please review, thanks.

jeandelavarene commented on java/odsa-spring/src/main/java/com/oracle/dev/jdbc/odsa/spring/Employee.java:
```diff
@@ -0,0 +1,118 @@
+/*
+  Copyright (c) 2021, 2022, Oracle and/or its affiliates.
```
Change year to 2024

juarezjuniorgithub replied:
Ok, I did it for all classes with the license note and updated them to 2024 as you requested

juarezjuniorgithub replied:
@Kuassim please review, approve and notify Jean per the process (2nd level approval). Thanks!

jeandelavarene commented on java/odsa-spring/src/main/java/com/oracle/dev/jdbc/odsa/spring/EmployeeNotFoundException.java:
```diff
@@ -0,0 +1,31 @@
+
+class EmployeeNotFoundException extends RuntimeException {
+
+  private static final long serialVersionUID = 7768215094663842819L;
```
Please check if this is needed.

juarezjuniorgithub replied:
@Kuassim please review, approve and notify Jean per the process (2nd level approval). Thanks!

Kuassim replied:
Approved; up to @jeandelavarene

jeandelavarene commented:
@juarezjuniorgithub please remove the target directory from your PR. You can add it to your .gitignore.
See this: https://stackoverflow.com/questions/991801/git-ignores-and-maven-targets
Also please upgrade the JDBC driver to 23.6.

juarezjuniorgithub commented:
Hi @Kuassim I have followed Jean's suggestion and updated the driver version to 23.6.0.24.10. 
However, let's try to abide by our latest agreements, as asking recurrently to update the driver version before the 
very first release of a given code sample configures a dead-end situation, where it's being kept in a situation of 
open modification. Such updates (versions/releases) will be done as part of our agreed Quarterly QA Freshness Pass activity. 
I have also addressed the issues with the /target directory as requested. 
@Kuassim please approve. Thanks

jeandelavarene commented on java/odsa-spring/README.md:
```diff
@@ -0,0 +1,2 @@
+# odsa-spring
+[]()
```
The README should at least mention what this code sample is about.
